src/db/delete.rs

```diff
@@ -119,50 +121,57 @@ fn delete_crate_from_database(conn: &Connection, name: &str, crate_id: i32) -> R
     Ok(())
 }
 
-fn delete_prefix_from_s3(s3: &S3Client, name: &str) -> Result<(), Error> {
-    let mut continuation_token = None;
-    loop {
-        let list = s3
-            .list_objects_v2(ListObjectsV2Request {
-                bucket: S3_BUCKET_NAME.into(),
-                prefix: Some(name.into()),
-                continuation_token,
-                ..ListObjectsV2Request::default()
-            })
-            .sync()?;
-
-        let to_delete = list
-            .contents
-            .unwrap_or_else(Vec::new)
-            .into_iter()
-            .filter_map(|o| o.key)
-            .map(|key| ObjectIdentifier {
-                key,
-                version_id: None,
-            })
-            .collect::<Vec<_>>();
-        let resp = s3
-            .delete_objects(DeleteObjectsRequest {
-                bucket: S3_BUCKET_NAME.into(),
-                delete: rusoto_s3::Delete {
-                    objects: to_delete,
-                    quiet: None,
-                },
-                ..DeleteObjectsRequest::default()
-            })
-            .sync()?;
-        if let Some(errs) = resp.errors {
-            for err in &errs {
-                log::error!("error deleting file from s3: {:?}", err);
+fn delete_prefix_from_s3(s3: &S3Backend, name: &str) -> Result<(), Error> {
```

Comment:
Now that we use a global `S3_RUNTIME`, we can change this back to just take the client. Same for various other places that changed to take a backend.